### DISCUSSION TITLE
Why HttpRoute is not included in OpenTelemetry.Instrumentation.Http metrics?

### DISCUSSION THREAD

**Discussion by 474D** [open-telemetry/opentelemetry-dotnet#4311] [Q&A] [answered]
Hi 👋, apologies if this question may sound silly, but I will give a try (I'm pretty new here)
I would like to ask why **HttpRoute** is not included in HttpClient instrumentation metrics?   
Is it by design? (in semantic conventions its only part of HttpServerMetrics).

For example making a call to **https://www.somegateway.com/some-resource**
will only provide a metric of **net_peer_name = www.somegateway.com**

Why not to extract **some-resource** and prefill _SemanticConventions.AttributeHttpRoute_ value?

**Comment by vishweshbankwar** [comment, marked as answer]
Hello, It is by design. HttpClient metric instrumentation only adds the attributes specified in the spec below.
https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/http-metrics.md#metric-httpclientduration
